### PR TITLE
Add script for installing dependencies

### PR DESCRIPTION
**Pull request by gunchleoc**
This script will try to detect the Linux distribution and install its dependencies.
Operating systems can also be specified manually on the command line.
The package lists were taken from the Wiki and might not be current for all distributions.

### CODE REVIEW

File: install-dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    mingw-w64-i686-glbinding mingw-w64-i686-glew
+elif [ "$DISTRO" == "msys64" ]; then
+   echo "Installing dependencies for 64-bit Windows..."
+   pacman -S mingw-w64-x86_64-toolchain git mingw-w64-x86_64-cmake \
+    mingw-w64-x86_64-ninja mingw-w64-x86_64-boost mingw-w64-x86_64-SDL2_ttf \
+    mingw-w64-x86_64-SDL2_mixer mingw-w64-x86_64-SDL2_image \
+    mingw-w64-x86_64-glbinding mingw-w64-x86_64-glew
+elif [ "$DISTRO" == "homebrew" ]; then
+   echo "Installing dependencies for Mac Homebrew..."
+   brew install boost bzrtools cmake doxygen gettext glew graphviz icu4c jpeg \
```
Comment (stonerl) [inline install-dependencies.sh]: `bzrtools` isn't necessary anymore.

File: install-dependencies.sh [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+# If distro was not given on the command line, try to autodetect
+if [ -z "$DISTRO" ]; then
+   echo "Autodetecting operating system..."
+   if [ -f /etc/fedora-release -o -f /etc/redhat-release -o -f /etc/centos-release ]; then
+      DISTRO="fedora"
+   elif [ -f /etc/gentoo-release ]; then
+      DISTRO="gentoo"
+   elif [ -f /etc/SuSE-release ]; then
+      DISTRO="suse"
+      echo "Installing dependencies for SuSE"
```
Comment (stonerl) [inline install-dependencies.sh]: Redundant since it is echoed during the installation.

File: install-dependencies.sh [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+   elif [ -f /etc/slackware-release -o -f /etc/slackware-version ]; then
+      DISTRO="slackware"
+   elif [ -f /etc/mandriva-release ]; then
+      DISTRO="mandriva"
+   elif [ -f /etc/debian_version ]; then
+      DISTRO="debian"
+   fi
```
Comment (stonerl) [inline install-dependencies.sh:83]:
```bash
elif [ -f /usr/local/Homebrew/bin/brew ]; then
   DISTRO="homebrew"
fi
```

**Review by stonerl — CHANGES_REQUESTED** [review]

File: install-dependencies.sh (hunk 1 above)
Comment (gunchleoc) [reply install-dependencies.sh]: What's the name of the corresponding git package?

File: install-dependencies.sh (hunk 2 above)
Comment (gunchleoc) [reply install-dependencies.sh]: I'll delete it.

File: install-dependencies.sh (hunk 3 above)
Comment (gunchleoc) [reply install-dependencies.sh:83]: Will add.

**Review by stonerl — APPROVED** [review]

**Pull request merged**